tools/pod_translator.py

```diff
@@ -0,0 +1,156 @@
+        ]
+
+        for file_name in pod_file_names:
+            with zip.open(file_name) as pod_content:
+                log_content = pod_content.read().decode("utf-8")
+                data.append(translate_pod(content=log_content))
+
+    # sort data by namespace and name
+    data.sort(key=lambda x: (x["namespace"], x["name"]))
+
+    output_ext = "json" if output_format == "json" else "txt"
+    output_template_path = f"./translated_pods.{output_ext}"
+
+    with open(output_template_path, "w") as write_file:
+        if output_format == "json":
+            write_file.write(json.dumps(data, indent=2))
+        elif output_format == "txt":
+            table = convert_to_table(data)
+            console = Console(file=write_file)
+            console.print(table)
+
+    # Test serialized content
+    with open(output_template_path, "r") as read_file:
+        content = read_file.read()
+
+    if output_format == "json":
+        payload_to_integrate = json.loads(content)
+        assert payload_to_integrate == data
+        TestCase().assertEqual(data, payload_to_integrate)
+    elif output_format == "txt":
+        payload_to_integrate = content
+        test_file_path = "./test_file.txt"
+        with open(test_file_path, "w") as write_file:
+            console = Console(file=write_file)
+            console.print(table)
+
+        with open(test_file_path, "r") as read_file:
+            expected_content = read_file.read()
+            assert payload_to_integrate == expected_content
+            TestCase().assertEqual(expected_content, payload_to_integrate)
+
+        # remove the test file
+        os.remove(test_file_path)
+
+    print("Expected data assertions passed!")
+
+
+def translate_pod(content: str):
+    # Load the YAML content
+    log = yaml.safe_load(content)
+
+    translated = {"name": log["metadata"]["name"], "namespace": log["metadata"]["namespace"]}
+    container_statuses = log["status"].get("containerStatuses", [])
+
+    pod_deletion_timestamp = log["metadata"].get("deletionTimestamp", None)
+    pod_phase = log["status"]["phase"]
+
+    # Determine pod status
+    if pod_deletion_timestamp and pod_phase not in ["failed", "succeeded"]:
+        translated["status"] = "Terminating"
+    else:
+        translated["status"] = pod_phase.capitalize()
+
+    readiness = sum(1 for status in container_statuses if status.get("ready"))
+    restart_count = sum(status.get("restartCount", 0) for status in container_statuses)
+
+    translated["restarts"] = restart_count
+    translated["containersReady"] = f"{readiness}/{len(container_statuses)}"
+
+    # Ensure the datetime object is in UTC
+    created_at = datetime.fromisoformat(log["metadata"]["creationTimestamp"]).replace(tzinfo=timezone.utc)
+
+    # get current time
+    current_time = datetime.fromisoformat(get_timestamp_now_utc()).replace(tzinfo=timezone.utc)
+    translated["age"] = str(current_time - created_at)
```

Comment:
I would not do this - it will provide invalid data based on how old the support bundle is. 

Is there a way to get the time the support bundle is created (ex: the support bundle name, assuming it is not changed - you can prob throw an error if the name has changed)